src/Config/RabbitMQExtensionConfigProvider.cs

```diff
@@ -51,27 +50,22 @@ public void Initialize(ExtensionConfigContext context)
             rule.AddOpenConverter<OpenType.Poco, byte[]>(typeof(PocoToBytesConverter<>));
 
             var triggerRule = context.AddBindingRule<RabbitMQTriggerAttribute>();
-            triggerRule.BindToTrigger<BasicDeliverEventArgs>(new RabbitMQTriggerAttributeBindingProvider(
+
+            // More details about why the BindToTrigger was chosen instead of BindToTrigger<BasicDeliverEventArgs> detailed here https://github.com/Azure/azure-functions-rabbitmq-extension/issues/110
+            triggerRule.BindToTrigger(new RabbitMQTriggerAttributeBindingProvider(
                     _nameResolver,
                     this,
                     _logger,
                     _options,
                     _configuration));
-
-            // Converts BasicDeliverEventArgs to string so user can extract received message.
-            triggerRule.AddConverter<BasicDeliverEventArgs, string>(args => Encoding.UTF8.GetString(args.Body))

```

Comment:
So you're now implementing your own conversions manually in the value binder, supporting string, BasicDeliverEventArgs, etc. How good is the E2E test coverage for all the various bindings that are supported currently? I'm wondering if these changes mean we're losing some built in conversion handling. E.g. I assume we currently support binding to string, JObject, Pocos, BasicDeliverEventArgs, Stream, TextReader, etc. - basically the built ins defined in ConverterManager https://github.com/Azure/azure-webjobs-sdk/blob/b798412ad74ba97cf2d85487ae8479f277bdd85c/src/Microsoft.Azure.WebJobs.Host/Bindings/ConverterManager.cs#L33. Need to make sure we're not breaking existing scenarios.

Reply:
I tested this out by binding with the afore mentioned types before and after this change. Binding to types like Stream, TextReader, TextWriter, byte[] did not work in the older version and remains the same with this change as well. Any complex types like that, RabbitMQ expects both the sender and the receiver to know the type-to-byte and byte-to-originalType conversion which is probably where binding to `BasicDeliverEventArgs` comes into picture since it retains the original content as is (I added special check for that as well).

Binding to string, JObject and Poco remains the same before and after this change as well. I verified this. If there's any other scenario I may have missed, please let me know.